subscription_web_access/models/res_users.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2019 Coop IT Easy SCRL fs
+#   Robin Keunen <robin@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+import json
+
+from odoo import api, models
+from odoo.exceptions import MissingError
+
+
+class ResUsers(models.Model):
+    _inherit = "res.users"
+
+    @api.model
+    def get_subscription(self, user_id):
+        user = self.env["res.users"].browse(user_id)
+        if not user:
+            return json.dumps({"error": "user not found"})
+
+        try:
+            partner = user.partner_id
+        except MissingError:
```

Comment:
will this be raised if `user.partner_id` is null? i think it will return an empty `res.partner` recordset.

Reply:
Indeed the error is not raised. I change it with an "if".
However this case should never happen, Odoo doesn't seem to allow users without related partners.